src/karmen_backend/server/clients/octoprint.py

```diff
@@ -94,80 +102,63 @@ def add_api_key(self, api_key):
         self.client_info.api_key = api_key
         self.http_session.headers.update({"X-Api-Key": api_key})
 
-    def _http_get(self, path, force=False, timeout=None):
```

Comment:
_http_get and _http_post shared most of the code. Therefor I created a `_request` method with the common part and cleand up `_http_get` and `_http_post` to reuse the `_reqeust` method.

I also added new argumetn `hide_errors` which does not raise exception in case of error but rather silently returns `None` for compatibility with the rest of code.